src/cnaas_nms/db/settings.py

```diff
@@ -683,3 +751,91 @@ def get_group_regex(group_name: str) -> Optional[str]:
+    """Return dicts with {name: priority} for groups"""
+    groups_priorities = {}
+
+    if not settings:
+        settings, _ = get_group_settings()
+    if settings is None:
+        return groups_priorities
+    if 'groups' not in settings:
+        return groups_priorities
+    if settings['groups'] is None:
+        return groups_priorities
+    for group in settings['groups']:
+        if 'name' not in group['group']:
+            continue
+        if 'group_priority' not in group['group'] or group['group']['group_priority'] == 0:
+            continue
+        if hostname:
+            if 'regex' not in group['group']:
+                continue
+            # TODO: try and catch, report what regex failed
+            if not re.match(group['group']['regex'], hostname):
+                continue
+        groups_priorities[group['group']['name']] = group['group']['group_priority']
+    return groups_priorities
+
+
+def get_groups_priorities_sorted(hostname: Optional[str] = None,
+                                 settings: Optional[dict] = None) -> Dict[str, int]:
+    return {k: v for k, v in sorted(get_groups_priorities(hostname, settings).items(),
+                                    key=lambda item: item[1],  # sort on value(priority)
+                                    reverse=True)}  # sort highest priority first
+
+
+def find_primary_group(secondary_groups: list,
+                       groups_priorities_sorted: Dict[str, int]) -> str:
+    for prio_group in groups_priorities_sorted.keys():
+        for sec_group in secondary_groups:
+            if prio_group == sec_group:
+                return prio_group
+    return 'DEFAULT'
+
+
+def parse_device_primary_groups() -> Dict[str, str]:
+    """Returns a dict with {hostname: primary_group} from settings"""
+    groups_priorities_sorted = get_groups_priorities_sorted()
+    device_primary_group: Dict[str, str] = {}
+    with sqla_session() as session:
+        devices: List[Device] = session.query(Device).all()
+        for dev in devices:
+            groups = get_groups(dev.hostname)
+            primary_group: str = find_primary_group(groups, groups_priorities_sorted)
+            device_primary_group[dev.hostname] = primary_group
+    return device_primary_group
+
+
+def update_device_primary_groups():
+    device_primary_group = parse_device_primary_groups()
+    if not device_primary_group:
+        return
+    with redis_session() as redis:
+        redis.hset("device_primary_group", mapping=device_primary_group)
+
+
+def get_device_primary_groups(no_cache: bool = False) -> Dict[str, str]:
+    """Returns a dict with {hostname: primary_group} from redis
+
+    Args:
+        no_cache: Update redis cache before returning data
+    """
+    logger = get_logger()
+    # update redis if redis is empty
+    with redis_session() as redis:
+        if not redis.exists("device_primary_group"):
+            no_cache = True
+    if no_cache:
```

Comment:
The lines above can be simplified:

```python
if not redis.exists("device_primary_group"):
    update_device_primary_groups()
```

Reply:
Yeah ok that might be easier to read yes :)